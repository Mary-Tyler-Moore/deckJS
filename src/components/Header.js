import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  Navbar,
  NavbarBrand,
  NavbarStart,
  NavbarEnd,
  NavbarItem,
  Container,
} from 'bloomer'
import SocialLinks from '../data/Social'

const LinkList = [
  {
    url: '/',
    title: 'Home',
  },
  {
    url: '/code',
    title: 'GitHub',
  },
]

const NavbarRoute = props => (
  <NavbarItem href={props.href}>
    {props.content}
  </NavbarItem>
)

const SocialMediaLinks = () => (
  <div style={{ display: 'flex' }}>
    {SocialLinks.filter(link => link.isPrimary).map(c => {
      return (
        <NavbarItem href={c.href} target="_blank" key={c.id}>
          <FontAwesomeIcon icon={c.classFontAwesome} fixedWidth />
        </NavbarItem>
      )
    })}
  </div>
)

const Header = () => (
  <Container>
    <Navbar>
      <NavbarBrand style={{ width: '100%' }} isDisplay="flex">
        <NavbarStart style={{ display: 'flex' }}>
          <SocialMediaLinks />
        </NavbarStart>
        <NavbarEnd style={{ display: 'flex', marginLeft: 'auto' }}>
          <NavbarRoute href="/" content="Home" />
          <NavbarRoute href="https://github.com/borjadotai/DeckJS" content="GitHub" />
        </NavbarEnd>
      </NavbarBrand>
    </Navbar>
  </Container>
)

export default Header
