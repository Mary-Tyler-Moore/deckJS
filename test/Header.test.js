import React from 'react'
import Header from '../src/components/Header'
import { shallow, mount } from 'enzyme'

describe('Header', () => {
  it('should render', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
  })
})