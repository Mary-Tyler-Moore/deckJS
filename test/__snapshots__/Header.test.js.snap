// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <withHelpersModifiers(Navbar)>
        <withHelpersModifiers(NavbarBrand)
          isDisplay="flex"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <withHelpersModifiers(NavbarStart)
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <SocialMediaLinks />
          </withHelpersModifiers(NavbarStart)>
          <withHelpersModifiers(NavbarEnd)
            style={
              Object {
                "display": "flex",
                "marginLeft": "auto",
              }
            }
          >
            <NavbarRoute
              content="Home"
              href="/"
            />
            <NavbarRoute
              content="GitHub"
              href="https://github.com/P3rzival/DeckJS"
            />
          </withHelpersModifiers(NavbarEnd)>
        </withHelpersModifiers(NavbarBrand)>
      </withHelpersModifiers(Navbar)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <withHelpersModifiers(NavbarBrand)
          isDisplay="flex"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <withHelpersModifiers(NavbarStart)
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <SocialMediaLinks />
          </withHelpersModifiers(NavbarStart)>
          <withHelpersModifiers(NavbarEnd)
            style={
              Object {
                "display": "flex",
                "marginLeft": "auto",
              }
            }
          >
            <NavbarRoute
              content="Home"
              href="/"
            />
            <NavbarRoute
              content="GitHub"
              href="https://github.com/P3rzival/DeckJS"
            />
          </withHelpersModifiers(NavbarEnd)>
        </withHelpersModifiers(NavbarBrand)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <withHelpersModifiers(NavbarStart)
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <SocialMediaLinks />
            </withHelpersModifiers(NavbarStart)>,
            <withHelpersModifiers(NavbarEnd)
              style={
                Object {
                  "display": "flex",
                  "marginLeft": "auto",
                }
              }
            >
              <NavbarRoute
                content="Home"
                href="/"
              />
              <NavbarRoute
                content="GitHub"
                href="https://github.com/P3rzival/DeckJS"
              />
            </withHelpersModifiers(NavbarEnd)>,
          ],
          "isDisplay": "flex",
          "style": Object {
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <SocialMediaLinks />,
              "style": Object {
                "display": "flex",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <NavbarRoute
                  content="Home"
                  href="/"
                />,
                <NavbarRoute
                  content="GitHub"
                  href="https://github.com/P3rzival/DeckJS"
                />,
              ],
              "style": Object {
                "display": "flex",
                "marginLeft": "auto",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "content": "Home",
                  "href": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "content": "GitHub",
                  "href": "https://github.com/P3rzival/DeckJS",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <withHelpersModifiers(Navbar)>
          <withHelpersModifiers(NavbarBrand)
            isDisplay="flex"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <withHelpersModifiers(NavbarStart)
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <SocialMediaLinks />
            </withHelpersModifiers(NavbarStart)>
            <withHelpersModifiers(NavbarEnd)
              style={
                Object {
                  "display": "flex",
                  "marginLeft": "auto",
                }
              }
            >
              <NavbarRoute
                content="Home"
                href="/"
              />
              <NavbarRoute
                content="GitHub"
                href="https://github.com/P3rzival/DeckJS"
              />
            </withHelpersModifiers(NavbarEnd)>
          </withHelpersModifiers(NavbarBrand)>
        </withHelpersModifiers(Navbar)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <withHelpersModifiers(NavbarBrand)
            isDisplay="flex"
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <withHelpersModifiers(NavbarStart)
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <SocialMediaLinks />
            </withHelpersModifiers(NavbarStart)>
            <withHelpersModifiers(NavbarEnd)
              style={
                Object {
                  "display": "flex",
                  "marginLeft": "auto",
                }
              }
            >
              <NavbarRoute
                content="Home"
                href="/"
              />
              <NavbarRoute
                content="GitHub"
                href="https://github.com/P3rzival/DeckJS"
              />
            </withHelpersModifiers(NavbarEnd)>
          </withHelpersModifiers(NavbarBrand)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <withHelpersModifiers(NavbarStart)
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <SocialMediaLinks />
              </withHelpersModifiers(NavbarStart)>,
              <withHelpersModifiers(NavbarEnd)
                style={
                  Object {
                    "display": "flex",
                    "marginLeft": "auto",
                  }
                }
              >
                <NavbarRoute
                  content="Home"
                  href="/"
                />
                <NavbarRoute
                  content="GitHub"
                  href="https://github.com/P3rzival/DeckJS"
                />
              </withHelpersModifiers(NavbarEnd)>,
            ],
            "isDisplay": "flex",
            "style": Object {
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <SocialMediaLinks />,
                "style": Object {
                  "display": "flex",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <NavbarRoute
                    content="Home"
                    href="/"
                  />,
                  <NavbarRoute
                    content="GitHub"
                    href="https://github.com/P3rzival/DeckJS"
                  />,
                ],
                "style": Object {
                  "display": "flex",
                  "marginLeft": "auto",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "content": "Home",
                    "href": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "content": "GitHub",
                    "href": "https://github.com/P3rzival/DeckJS",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
